### PR TITLE
Used cached toolchain containers by default.

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

In 2.0 `CLEAN_TOOLCHAIN_CONTAINERS` defaults to `y` unless `QUICK_REBUILD_TOOLCHAIN=y` is set. This means that most of the time when running `make clean` we will remove all the build containers associated with the current checkout (they are identified via the docker variable `MARINER_BUILD_DIR` which is included in each image).

The containers are tagged with a hash of the interesting input components (`./container/toolchain-sha256sums` `./container/toolchain_build_in_chroot.sh` `./create_toolchain_in_container.sh`). This means if a branch changes any of these files the previous containers will not be valid and the bootstrap will be re-run.

The bootstrap container is generally extremely stable, which means this build should be a very uncommon event. Unfortunately, with the current behavior of `make clean` these containers are discarded very easily (you can use `make clean CLEAN_TOOLCHAIN_CONTAINERS=n`, but this feels unnatural). 

With his change, globally setting `QUICK_REBUILD=y` becomes a valid strategy for building (and maybe that can be set to on by default as well?). Prior to this using that flag would often result in several hours wasted rebuilding the bootstrap toolchain *even if we could have downloaded all the toolchain rpms*.

Now at least we will skip over the bootstrap quickly and the incremental toolchain can build very fast if all rpms are available in the repo.

###### Change Log  

- Default `CLEAN_TOOLCHAIN_CONTAINERS=n` 

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local builds

### CODE REVIEW

**Review by jslobodzian — APPROVED** [review]

**Review by PawelWMS — APPROVED** [review]

File: toolkit/scripts/toolchain/create_toolchain_in_container.sh [hunk 1]
```diff
@@ -12,7 +12,17 @@ INCREMENTAL_TOOLCHAIN=$4
 ARCHIVE_TOOL=$5
 
 # Grab an identity for the raw toolchain components so we can avoid rebuilding it if it hasn't changed
-sha_component_tag=$(sha256sum ./container/toolchain-sha256sums ./container/toolchain_build_in_chroot.sh ./create_toolchain_in_container.sh | sha256sum | cut -d' ' -f1 )
+files_to_watch=( "./create_toolchain_in_container.sh" \
+                    "./container/toolchain-sha256sums" \
```
Comment (anphel31) [inline toolkit/scripts/toolchain/create_toolchain_in_container.sh:16]: Can you also add `toolchain-remote-wget-list`?

Comment (anphel31) [reply toolkit/scripts/toolchain/create_toolchain_in_container.sh:16]: and please re-alphabetize the list

**Review by anphel31 — APPROVED** [review]

File: toolkit/scripts/toolchain/create_toolchain_in_container.sh (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/scripts/toolchain/create_toolchain_in_container.sh]: done

**Comment by dmcilvaney** [comment]
Force rebuild: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=528368&view=results
No rebuild:     https://dev.azure.com/mariner-org/mariner/_build/results?buildId=527997&view=results

**Comment by dmcilvaney** [comment]
3.0 builds always regenerate raw toolchain, both builds seem ok though.

**Pull request merged**